ui_components/src/ramen/RamenShop.js

```diff
@@ -0,0 +1,231 @@
+import { apiFetch } from "../utils/network.js";
+import { ResourceBar } from "../utils/resourceBar.js";
+import { parseKeywords } from "../utils/parseKeywords.js";
+
+function RamenShopReactContainer({
+    ramenShopAPI,
+    playerData,
+    playerResourcesData,
+    characterRamenData,
+    ramenOwnerDetails,
+    basicRamenOptions,
+    specialRamenOptions,
+    mysteryRamenDetails
+}) {
+    return (
+        <ModalProvider>
+            <RamenShop
+                ramenShopAPI={ramenShopAPI}
+                playerData={playerData}
+                playerResourcesData={playerResourcesData}
+                characterRamenData={characterRamenData}
+                ramenOwnerDetails={ramenOwnerDetails}
+                basicMenuOptions={basicRamenOptions}
+                specialMenuOptions={specialRamenOptions}
+                mysteryRamenDetails={mysteryRamenDetails}
+            />
+        </ModalProvider>
+    );
+}
+
+function RamenShop({
+    ramenShopAPI,
+    playerData,
+    playerResourcesData,
+    characterRamenData,
+    ramenOwnerDetails,
+    basicMenuOptions,
+    specialMenuOptions,
+    mysteryRamenDetails,
+}) {
+    const [playerDataState, setPlayerDataState] = React.useState(playerData);
+    const [playerResourcesDataState, setPlayerResourcesDataState] = React.useState(playerResourcesData);
+    const [characterRamenDataState, setCharacterRamenDataState] = React.useState(characterRamenData);
+    const [mysteryRamenDetailsState, setMysteryRamenDetailsState] = React.useState(mysteryRamenDetails);
+    const { openModal } = useModal();
+    function BasicRamen({ index, ramenInfo, PurchaseBasicRamen }) {
+        return (
+            <div key={index} className="basic_ramen">
+                <img src={ramenInfo.image} className="basic_ramen_img" />
+                <div className="basic_ramen_details">
+                    <div className="basic_ramen_name">{ramenInfo.label}</div>
+                    <div className="basic_ramen_effect">{ramenInfo.health_amount} HP</div>
+                    <div className="basic_ramen_button" onClick={() => PurchaseBasicRamen(ramenInfo.ramen_key)}><>&yen;</>{ramenInfo.cost}</div>
+                </div>
+            </div>
+        );
+    }
+    function SpecialRamen({ index, ramenInfo }) {
+        return (
+            <div key={index} className="special_ramen">
+                <img src={ramenInfo.image} className="special_ramen_img" />
+                <div className="special_ramen_name">{ramenInfo.label}</div>
+                <div className="special_ramen_description">{ramenInfo.description}</div>
+                <div className="special_ramen_effect">{ramenInfo.effect}</div>
+                <div className="special_ramen_duration">Duration: {ramenInfo.duration} minutes</div>
+                <div className="special_ramen_button" onClick={() => PurchaseSpecialRamen(ramenInfo.ramen_key)}><>&yen;</>{ramenInfo.cost}</div>
+            </div>
+        );
+    }
```

Comment:
These are their own components, please move out of the body of the `RamenShop` component and declare below